bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoHandler.java

```diff
@@ -111,63 +99,107 @@ public WemoHandler(Thing thing, UpnpIOService upnpIOService, WemoHttpCall wemoHt
     public void initialize() {
         Configuration configuration = getConfig();
 
-        if (configuration.get("udn") != null) {
-            logger.debug("Initializing WemoHandler for UDN '{}'", configuration.get("udn"));
-            service.registerParticipant(this);
-            onSubscription();
-            onUpdate();
+        if (configuration.get(UDN) != null) {
+            logger.debug("Initializing WemoHandler for UDN '{}'", configuration.get(UDN));
+            UpnpIOService localService = service;
+            if (localService != null) {
+                localService.registerParticipant(this);
+            }
+            host = getHost();
+            pollingJob = scheduler.scheduleWithFixedDelay(this::poll, 0, DEFAULT_REFRESH_INTERVALL_SECONDS,
+                    TimeUnit.SECONDS);
             updateStatus(ThingStatus.ONLINE);
         } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/config-status.error.missing-udn");
             logger.debug("Cannot initalize WemoHandler. UDN not set.");
         }
     }
 
     @Override
     public void dispose() {
-        logger.debug("WeMoHandler disposed.");
+        logger.debug("WemoHandler disposed for thing {}", getThing().getUID());
 
-        ScheduledFuture<?> job = refreshJob;
-        if (job != null && !job.isCancelled()) {
+        ScheduledFuture<?> job = this.pollingJob;
+        if (job != null) {
             job.cancel(true);
         }
-        refreshJob = null;
+        this.pollingJob = null;
         removeSubscription();
+        UpnpIOService localService = service;
+        if (localService != null) {
+            localService.unregisterParticipant(this);
+        }
```

Comment:
Why? This is already done by `removeSubscription()` called from the line above?

Reply:
done